textpipe/pipeline.py

```diff
@@ -47,13 +55,14 @@ def save(self, filename):
         filename: location of where to serialize pipeline
 
         >>> filename = "test.json"
-        >>> Pipeline(['n_sents', 'clean_text']).save(filename)
+        >>> Pipeline(['Nsents', 'CleanText']).save(filename)
         >>> sorted(json.load(open(filename)).items())
-        [('hint_language', None), ('language', None), ('pipeline', ['n_sents', 'clean_text'])]
+        [('hint_language', None), ('kwargs', {}), ('language', None), ('operations', ['Nsents', 'CleanText'])]
         """
-
         with open(filename, 'w') as json_file:
-            json.dump(self.__dict__, json_file)
+            # serialize all public attrs
+            serialize = dict(filter(lambda i: not i[0].startswith('_'), self.__dict__.items()))
```

Comment:
`serialized = {k: v for k, v in self.__dict__.items() if not k.startswith('_')}`

Reply:
@dodijk 's non-lambda approach reads much cleaner